Add classic project (#240)
plone/cookieplone-templates: templates/projects/classic/{{ cookiecutter.__folder_name }}/README.md

# {{ cookiecutter.title }} 🚀

[![Built with Cookieplone](https://img.shields.io/badge/built%20with-Cookieplone-0083be.svg?logo=cookiecutter)](https://github.com/plone/cookieplone-templates/)
[![Black code style](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
[![Backend Tests]({{ cookiecutter.__repository_url }}/actions/workflows/backend.yml/badge.svg)]({{ cookiecutter.__repository_url }}/actions/workflows/backend.yml)

{{ cookiecutter.description }}

> ## Quick Start 🏁

Reviewer: ```suggestion
## Quick start 🏁
```

### Prerequisites ✅

-   An [operating system](https://6.docs.plone.org/install/create-project-cookieplone.html#prerequisites-for-installation) that runs all the requirements mentioned.
-   [uv](https://6.docs.plone.org/install/create-project-cookieplone.html#uv)
-   [Make](https://6.docs.plone.org/install/create-project-cookieplone.html#make)
-   [Git](https://6.docs.plone.org/install/create-project-cookieplone.html#git)
-   [Docker](https://docs.docker.com/get-started/get-docker/) (optional)

### Installation 🔧

1.  Clone this repository, then change your working directory.

    ```shell
    git clone {{ cookiecutter.__repository_git }}.git
    cd {{ cookiecutter.__project_slug }}
    ```

2.  Install this code base.

    ```shell
    make install
    ```

### Fire Up the Servers 🔥

1.  Create a new Plone site on your first run.

    ```shell
    make backend-create-site
    ```

2.  Start the backend at http://localhost:8080/.

    ```shell
    make backend-start
    ```

Voila! Your Plone site should be live and kicking! 🎉

### Local Stack Deployment 📦

> Deploy a local `Docker Compose` environment that includes:

Reviewer: ```suggestion
Deploy a local Docker Compose environment that includes the following.
```

> - Docker image for Backend 🖼️
> - A stack with a Traefik router and a Postgres database 🗃️
> - Accessible at [http://{{ cookiecutter.__project_slug }}.localhost](http://{{ cookiecutter.__project_slug }}.localhost) 🌐

Reviewer: The emojis are too much.

```suggestion
-   A Docker image for the backend
-   A stack with a Traefik router and a PostgreSQL database
-   Is accessible at [http://{{ cookiecutter.__project_slug }}.localhost](http://{{ cookiecutter.__project_slug }}.localhost)
```

> Execute the following:

Reviewer: ```suggestion
Run the following commands in a shell session.
```

```shell
make stack-start
make stack-create-site
```

And... you're all set! Your Plone site is up and running locally! 🚀

> ## Project Structure 🏗️

Reviewer: ```suggestion
## Project structure 🏗️
```

This monorepo consists of the following distinct sections:

- **backend**: Houses the API and Plone installation, utilizing pip instead of buildout, and includes a policy package named {{ cookiecutter.python_package_name }}.

> - **devops**: Encompasses Docker Stack, Ansible playbooks, and Cache settings.

Reviewer: ```suggestion
- **devops**: Encompasses Docker stack, Ansible playbooks, and cache settings.
```

- **docs**: Scaffold for writing documentation for your project.

> ### Why This Structure? 🤔

Reviewer: ```suggestion
### Why this structure? 🤔
```

> - All necessary codebases to run the site are contained within the repo (excluding existing addons for Plone and React).
> - Specific GitHub Workflows are triggered based on changes in each codebase (refer to .github/workflows).
> - Simplifies the creation of Docker images for each codebase.
> - Demonstrates Plone installation/setup without buildout.

Reviewer: ```suggestion
-   All necessary code bases to run a Plone site are contained within the repository, excluding existing add-ons for Plone and React.
-   Specific GitHub Workflows are triggered based on changes in each code base. Refer to the files in the directory `.github/workflows/`.
-   Simplifies the creation of Docker images for each code base.
-   Demonstrates Plone installation and setup without buildout.
```

> ## Code Quality Assurance 🧐

Reviewer: ```suggestion
## Code quality assurance 🧐
```

> To automatically format your code and ensure it adheres to quality standards, execute:

Reviewer: ```suggestion
To check your code against quality standards, run the following shell command.
```

```shell
make check
```

> ### Format the codebase
> 
> To format the codebase, it is possible to run `format`:

Reviewer: ```suggestion
### Format the code base

To format and rewrite the code base, ensuring it adheres to quality standards, run the following shell command.
```

```shell
make format
```

| Section | Tool | Description | Configuration |
| --- | --- | --- | --- |
| backend | Ruff | Python code formatting, imports sorting  | [`backend/pyproject.toml`](./backend/pyproject.toml) |
| backend | `zpretty` | XML and ZCML formatting  | -- |

### Linting the codebase
or `lint`:

 ```shell
make lint
```

| Section | Tool | Description | Configuration |
| --- | --- | --- | --- |
| backend | Ruff | Checks code formatting, imports sorting  | [`backend/pyproject.toml`](./backend/pyproject.toml) |
| backend | Pyroma | Checks Python package metadata  | -- |
| backend | check-python-versions | Checks Python version information  | -- |
| backend | `zpretty` | Checks XML and ZCML formatting  | -- |

## Internationalization 🌐

Generate translation files for Plone with ease:

```shell
make i18n
```

> ## Credits and Acknowledgements 🙏

Reviewer: ```suggestion
## Credits and acknowledgements 🙏
```
